controllers/openstackbaremetalset_controller.go

```diff
@@ -872,7 +888,7 @@ func (r *OpenStackBaremetalSetReconciler) baremetalHostProvision(
 				instance.Status.BaremetalHosts[bmhStatus.Hostname] = bmhStatus
 
 				// update status with host assignment
-				if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
+				if err := r.Status().Update(context.Background(), instance); err != nil {
```

Comment:
Curious as to why we're using `Background()` here instead of the passed-in `ctx`

Reply:
as discussed on slack, I just wanted to make sure in some places to use a context which It is never canceled, but since we only support a single instance of the operator it probably does not make a different right now.